src/components/Tables/components/TableActionMenu.tsx

```diff
@@ -0,0 +1,59 @@
+import { DropdownMenu } from '@radix-ui/themes'
+import { DotsVerticalIcon } from '@radix-ui/react-icons'
+import { useState } from 'react'
+import { ConfirmDelete } from './ConfirmDelete'
+
+type Props = {
+    isFeatured: boolean
+    onToggleFeatured: () => void
+    onEdit: () => void
+    onDelete: () => void
+}
+
+export function TableActionMenu({
+    isFeatured,
+    onToggleFeatured,
+    onEdit,
+    onDelete
+}: Props) {
+    const [deleteDialogOpen, setDeleteDialogOpen] = useState(false)
+
+    return (
+        <>
+            <DropdownMenu.Root>
+                <DropdownMenu.Trigger>
+                    <DotsVerticalIcon />
+                </DropdownMenu.Trigger>
+                <DropdownMenu.Content size="2">
+                    <DropdownMenu.Item onClick={onEdit}>Edit</DropdownMenu.Item>
+                    <DropdownMenu.Separator />
+                    <DropdownMenu.Item
+                        color={isFeatured ? undefined : 'green'}
+                        disabled={isFeatured}
+                        onClick={() => {
```

Comment:
Instead of having the arrow function here, you can just pass `onClick={onToggleFeatured}`